src/App.tsx

```diff
@@ -31,65 +32,128 @@ export function getReorderedGoods(
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handelSortAlphabetically = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
+
+  handelSortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  handelReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handelReset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const sortedGoods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={cn(
+              'button is-info is-large',
+              { 'is-light': sortType !== SortType.ALPHABET },
+            )}
+            onClick={this.handelSortAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={cn(
+              'button is-success is-large',
+              { 'is-light': sortType !== SortType.LENGTH },
+            )}
+            onClick={this.handelSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={cn(
+              'button is-warning is-large',
+              { 'is-light': !isReversed },
+            )}
+            onClick={this.handelReverse}
+          >
+            Reverse
+          </button>
+
+          {(isReversed || sortType !== SortType.NONE)
+            ? (
+              <button
+                type="button"
+                className="button is-danger is-large is-light"
+                onClick={this.handelReset}
+              >
+                Reset
+              </button>
+            )
+            : null}
```

Comment:
you shouldn't render it if you haven't

(isReversed || sortType !== SortType.NONE

try to use {(isReversed || sortType !== SortType.NONE) && (...